ouroboros-consensus/test-consensus/Test/Consensus/HardFork/Combinator/A.hs

```diff
@@ -296,69 +282,10 @@ instance LedgerSupportsMempool BlockA where
-
-              -- The ledger must report the scheduled transition to the next
-              -- era as soon as the block containing this transaction is
-              -- immutable (that is, at least @k@ blocks have come after) --
-              -- this happens elsewhere in the corresponding 'SingleEraBlock'
-              -- instance. It must not report it sooner than that because the
-              -- consensus layer requires that conversions about time (when
-              -- successful) must not be subject to rollback.
-              --
-              -- Consensus /also/ insists that as long as the transition to the
-              -- next era is not yet known (ie not yet determined by an
-              -- immutable block), there is a safe zone that extends past the
-              -- tip of the ledger in which we guarantee the next era will not
-              -- begin. This means that we must have an additional
-              -- @safeFromTipA k@ blocks /after/ reporting the transition and
-              -- /before/ the start of the next era.
-              --
-              -- Thus, we schedule the next era to begin with the first
-              -- upcoming epoch that starts /after/ we're guaranteed to see
-              -- both the aforementioned @k@ additional blocks and also a
-              -- further @safeFromTipA k@ slots after the last of those.
-
-              -- The last slot that must be in the current era
-              firstPossibleLastSlotThisEra =
-                  History.addSlots (stabilityWindowA k + safeFromTipA k) sno
-
-              -- The 'EpochNo' corresponding to 'firstPossibleLastSlotThisEra'
-              --
-              -- NOTE: We cannot use 'EpochInfo' for this. The 'EpochInfo'
-              -- only allows us to look ahead as far as the safe zone allows,
-              -- but here we are, almost by definition, looking further ahead
-              -- than that (stability window + safe zone).
-              --
-              -- (Once we construct an 'EpochInfo' with the known transition, it
-              -- would allow us to look this far ahead, but we cannot use it
-              -- to /determine/ the transition point.)
-              lastEpochThisEra =
-                  History.addEpochs
-                    (History.countSlots
-                       firstPossibleLastSlotThisEra firstSlotTipEpoch
-                     `div` unEpochSize epochSizeTipEpoch)
-                    tipEpoch
-
-              -- The first epoch that may be in the next era (recall: eras are
-              -- epoch-aligned)
-              firstEpochNextEra = succ lastEpochThisEra
-
-          return $ Ticked sno $ st {
-              lgrA_transition = Just (sno, firstEpochNextEra)
-            }
-    where
-      k = lcfgA_k ledgerConfig
-
-      runEI :: (EpochInfo Identity -> a -> Identity b) -> a -> b
-      runEI f x = runIdentity $ f ei x
+          return $ Ticked sno $ st { lgrA_transition = Just sno }
```

Comment:
While you're in here, I was tempted earlier to use the `Latest` monoid if only for indicativeness. Just a thought.